### PR TITLE
test: fix selenium error handling of libraries

### PR DESCRIPTION
**Pull request by jscotka**
failing and screenhot tested with one broken test. worked well in CI, commit with failed test removed
see log: 
https://fedorapeople.org/groups/cockpit/logs/pull-10731-20181205-134141-754ff9f1-selenium-firefox/log.html

### CODE REVIEW

File: test/avocado/testlib_avocado/seleniumlib.py [hunk 1]
```diff
@@ -56,6 +58,9 @@ class SeleniumTest(Test):
     :avocado: disable
     """
     def setUp(self):
+        # ensure that setup phase passed, otherwise handle it in tierDown
```
Comment (martinpitt) [inline test/avocado/testlib_avocado/seleniumlib.py]: As you consistently misspell this: It's "tear down", not "tier" (also in commit message). :smiley:

File: test/avocado/testlib_avocado/seleniumlib.py [hunk 2]
```diff
@@ -105,39 +110,50 @@ def connect_browser():
...
+            datesuffix = str(time.clock())[2:]
             try:
-                # use time.clock() to ensure that snapshot files are unique and ordered
-                # sample name is like: screenshot-teardown-172434.png
-                screenshot_file = "screenshotTeardown%s.png" % str(time.clock())[2:]
-
-                self.driver.save_screenshot(os.path.join(actualpath,screenshot_file))
-                self.log.error("Screenshot(teardown) - Wrote: " + screenshot_file)
+                stackinfo = [x[3] for x in inspect.stack() if x[3].startswith("test") or x[3] in ["tearDown", "setUp"]]
+            except Exception:
```
Comment (martinpitt) [inline test/avocado/testlib_avocado/seleniumlib.py]: This is a blanket exception handler again, which would also catch things like `AttributeError` or `NameError`. As the point of this PR is to clean up error handling, let's please not introduce new ones.

Comment (jscotka) [reply test/avocado/testlib_avocado/seleniumlib.py]: @martinpitt I understand, but the trouble is that I were unable to raise any exception. It is somehow connected that stack info is missing, but i have no idea if it raise any exception. So I can use there this NameError and attribute one for sure, seems these could be good ones.

Comment (martinpitt) [reply test/avocado/testlib_avocado/seleniumlib.py]: I'm afraid I don't understand this. But then again I also don't understand what kind of Exception you would possibly expect here in the first place. E. g. you *never* want to intercept `AttributeError` or `NameError`, these are usually typos in your program that blanket exception catching hides. It is possible to have a `KeyError` if you expect these arrey indexes to sometimes not exist, or `inspect.stack()` might fail.

Did you try what happens if you don't catch exceptions here at all?

Comment (jscotka) [reply test/avocado/testlib_avocado/seleniumlib.py]: I've looked to doc and it seems that there is no exception raised, just Null if not supported. so I'll repair it

**Pull request merged**